openedx/core/djangoapps/call_stack_manager/tests/tests.py

```diff
@@ -0,0 +1,188 @@
+"""
+Test cases for Call Stack Manager
+"""
+from mock import patch
+from .models import ModelMixin, ModelNothing, ModelMixinCSM, ModelAnotherCSM, ModelWithCSM, ModelWithCSMChild
+from django.test import TestCase
+from openedx.core.djangoapps.call_stack_manager import donottrack
+
+
+class TestingCallStackManager(TestCase):
+    """
+    Tests for call_stack_manager package
+    """
+    @patch('openedx.core.djangoapps.call_stack_manager.core.log.info')
+    def test_save(self, log_capt):
+        """ tests save functionality of call stack manager/ applies same for delete()
+        1. classes with CallStackMixin should participate in logging.
+        """
+        ModelMixin(id_field=1).save()
+        desired_class = ModelMixin
+        self.assertEqual(desired_class, log_capt.call_args[0][1])
+
+    @patch('openedx.core.djangoapps.call_stack_manager.core.log.info')
+    def test_withoutmixin_save(self, log_capt):
+        """tests save functionality of call stack manager / applies same for delete()
+        1. classes without CallStackMixin should not participate in logging
+        """
+        ModelAnotherCSM(id_field=1).save()
+        self.assertEqual(len(log_capt.call_args_list), 0)
+
+    @patch('openedx.core.djangoapps.call_stack_manager.core.log.info')
+    def test_queryset(self, log_capt):
+        """ Tests for Overriding QuerySet API
+        1. Classes with CallStackManager gets logged.
+        """
+        ModelMixinCSM(id_field=1).save()
+        ModelMixinCSM.objects.all()
+        desired_class = ModelMixinCSM
+        self.assertEqual(desired_class, log_capt.call_args[0][1])
```

Comment:
Should this check both the first and the second calls to `log.info`, since both `.save` and `.objects.all` should be captured?